src/stratis_cli/_errors.py

```diff
@@ -30,6 +30,83 @@ class StratisCliRuntimeError(StratisCliError):
     """
 
 
+class StratisNoChangeError(StratisCliError):
+    """
+    Raised if there was a failure due to no changed state in stratisd's engine.
+    """
+
+    def __init__(self, command, resources):
+        """ Initializer.
+
+            :param str command: the executed command
+            :param str or list of str resources: the target resources
+        """
+        super().__init__()
+        self.command = command
+        self.resources = resources
+
+    def __str__(self):
+        return "The command '{}' has already been completed for resource '{}'".format(
+            self.command, self.resources
+        )
+
+
+class StratisPartialChangeError(StratisCliError):
+    """
+    Raised if there was a failure due to partially changed state in stratisd's engine.
+    """
+
+    def __init__(self, command, changed_resources, unchanged_resources):
+        """ Initializer.
+
+            :param str command: the command run that caused the error
+            :param list of str changed_resources: the target resources that would change
+            :param list of str unchanged_resources: the target resources that would not change
+        """
+        super().__init__()
+        self.command = command
+        self.changed_resources = changed_resources
+        self.unchanged_resources = unchanged_resources
+
+    def __str__(self):
+        msg = "The command '{}' has already been completed for resources {} ".format(
+            self.command, self.unchanged_resources
+        )
+        if self.changed_resources != []:
+            msg += "but not yet for resources {}".format(self.changed_resources)
+        return msg
+
+
+class StratisInUseError(StratisCliError):
+    """
+    Raised if a block device is added as both a blockdev and a cachedev
+    """
+
+    def __init__(self, blockdevs, added_as="cache"):
+        """ Initializer.
+
+            :param list of str blockdevs: the blockdevs that would be added in both tiers
+            :param str added_as: whether the action causing the error on the cache or data tier
+        """
+        super().__init__()
```

Comment:
Do not call the initializer...use the ```#pylint: disable``` directive instead. Call the initializer and the auto-generated ```__repr__``` is incorrect. I'm not sure why this is, but this is definitely the case and appears to be the original motivation for the way the other exceptions were set up.

Reply:
Resolved.